soda/snowflake/soda/data_sources/snowflake_data_source.py

```diff
@@ -119,8 +119,14 @@ def sql_get_table_names_with_count(
         table_filter_expression = self.sql_table_include_exclude_filter(
             "table_name", "row_count", include_tables, exclude_tables
         )
-        where_clause = f"\nWHERE {table_filter_expression} \n" if table_filter_expression else ""
-        return f"SELECT table_name, row_count \n" f"FROM information_schema.tables" f"{where_clause}"
+        where_clause = f"AND {table_filter_expression}" if table_filter_expression else ""
+        sql = f"""
+            SELECT table_name, row_count
+            FROM information_schema.tables
+            WHERE table_schema != 'INFORMATION_SCHEMA'
```

Comment:
cannot this be the other way around, get metadata from only the desired db and schema? if so, then doing
```
self.schema = connection_properties.get("schema")
self.database = connection_properties.get("database")
```
in the `connect` method should do the trick, schema is picked up automatically in the `sql_table_include_exclude_filter` method and where clause for the schema is added.

If you are ok with this I can take care of it.